src/fsharp/CheckFormatStrings.fs

```diff
@@ -186,7 +186,7 @@ let parseFormatStringInternal (m: range) (fragRanges: range list) (g: TcGlobals)
        
        // Check if we've moved into the next fragment.  Note this will always activate on
        // the first step, i.e. when i=0
-       let (struct (fragLine, fragCol, fragments)) =
```

Comment:
I don't particularly like the way we allow `struct` in a pattern without parentheses.  But I guess it's ok to remove, just mentioning it

Reply:
If it's possible to change the tooling to always require parens around struct tuple patterns that would be good, though doesn't have to be part of this PR

Reply:
I honestly really like the fact struct itself can have parens omitted, struct tuples would otherwise be even heavier syntactically...

Reply:
What about using them in other tuples? Would you prefer extra parens too?

```fsharp
let f (a, struct (b, c)) = ...
```

Reply:
No not there.  I guess it's only in some positions, I'm not sure exactly which precedence level I'd expect parens to kick in